### PR TITLE
Logged per day chart

### PR DESCRIPTION
**Pull request by art1990**
Basically creating a LoggedPerDay chart(api, logic) + refactoring code

### CODE REVIEW

File: src/components/BarChart/StackedBarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
+import React from "react";
+import { View } from "react-native";
+import { Svg, Rect, G, Text } from "react-native-svg";
+import { AbstractChart } from "react-native-chart-kit";
+
+const barWidth = 32;
```
Comment (DimaMakarenko) [inline src/components/BarChart/StackedBarChart.tsx:9]: norm

File: src/components/BarChart/StackedBarChart.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    return barPercentage;
+  };
+
+  getBarRadius = (ret, x) => {
+    return this.props.chartConfig.barRadius && ret.length === x.length - 1
+      ? this.props.chartConfig.barRadius
+      : 0;
+  };
+
+  renderBars = (config) => {
```
Comment (CheeBam) [inline src/components/BarChart/StackedBarChart.tsx:23]:
```
renderBars = ({
      data,
      width,
      height,
      paddingTop,
      paddingRight,
      border,
      colors,
      stackedBar = false,
    }) => data.map((x, i) => { ...
```

File: src/components/BarChart/StackedBarChart.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+          />
+        );
+        st -= h;
+      }
+
+      return ret;
+    });
+  };
+
+  renderLegend = (config) => {
```
Comment (CheeBam) [inline src/components/BarChart/StackedBarChart.tsx:70]:
```
renderBars = ({legend, colors, width, height}) => legend.map((x, i) => { ...
```

File: src/components/BarChart/StackedBarChart.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      height,
+    };
+    let border = 0;
+    for (let i = 0; i < data.data.length; i++) {
+      const actual = data.data[i].reduce((pv, cv) => pv + cv, 0);
+      if (actual > border) {
+        border = actual;
+      }
+    }
+    var stackedBar = data.legend && data.legend.length == 0 ? false : true;
```
Comment (CheeBam) [inline src/components/BarChart/StackedBarChart.tsx:123]: ```const stackedBar = !(data.legend && data.legend.length == 0)```

File: src/components/IconButton/index.tsx [hunk 5]
```diff
@@ -74,17 +83,19 @@ const IconButton: React.FC<IIconButtonProps> = ({
...
   const iconSize = icon === "cansel" ? 12 : size;
 
   return (
     <ICButton
+      {...rest}
       icon={() => <Icon width={iconSize} heigth={iconSize} color={color} />}
       onPress={onPress}
       style={[
-        icon === "cansel" && {
+        ["cansel", "prev", "next"].indexOf(icon) !== -1 && {
```
Comment (CheeBam) [inline src/components/IconButton/index.tsx:98]: can**c**el

Comment (CheeBam) [reply src/components/IconButton/index.tsx:98]: use includes

File: src/components/BarChart/StackedBarChart.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+import React from "react";
+import { View } from "react-native";
+// svg
+import { Svg, Rect, G, Text } from "react-native-svg";
+// chart-kit
+import { AbstractChart } from "react-native-chart-kit";
+
+const barWidth = 32;
+
+class StackedBarChart extends AbstractChart {
```
Comment (CheeBam) [inline src/components/BarChart/StackedBarChart.tsx:11]: better to use everything in the same writing style, so use functional component

File: src/types/index.tsx [hunk 7]
```diff
@@ -121,6 +130,50 @@ export interface IWorkingTaskInfoProps {
...
+}
+
+export interface ITaskSwipeableInfoProps extends ITaskInfoProps {
+  onRemovePress: () => void;
+  onEditPress: () => void;
+  onResumePress: () => void;
+}
+
+export interface ITaskViewProps {
+  task: any;
```
Comment (CheeBam) [inline src/types/index.tsx:172]: any?

File: src/types/index.tsx [hunk 8]
```diff
@@ -50,6 +51,14 @@ export interface IInputProps {
   keyboardType?: KeyboardTypeOptions;
 }
 
+export interface IInputWithIconProps extends IInputProps {
+  style?: string;
+  inputStyle?: string | [] | {};
```
Comment (DimaMakarenko) [inline src/types/index.tsx:56]: delete string

File: src/components/BarChart/index.tsx [hunk 9]
```diff
@@ -1,57 +1,80 @@
...
-
 const chartConfig = {
-  backgroundColor: "red",
   backgroundGradientFrom: "#fff",
   backgroundGradientTo: "#fff",
-  decimalPlaces: 2, // optional, defaults to 2dp
+  // optional, defaults to 2dp
   color: (opacity = 1) => `rgba(0, 0, 0, 1)`,
-  labelColor: (opacity = 1) => `rgba(0, 0, 0, 0.4)`,
+  labelColor: (opacity = 1) => Colors.chartLabelColor,
```
Comment (DimaMakarenko) [inline src/components/BarChart/index.tsx:19]: not use opacity

**Pull request merged**